extensions/extended-isupport.md

```diff
@@ -0,0 +1,93 @@
+---
+title: "`extended-isupport` Extension"
+layout: spec
+copyrights:
+  -
+    name: "Simon Ser"
+    period: "2024"
+    email: "contact@emersion.fr"
+---
+
+## Notes for implementing work-in-progress version
+
+This is a work-in-progress specification.
+
+Software implementing this work-in-progress specification MUST NOT use the
+unprefixed `extended-isupport` capability name. Instead, implementations SHOULD
+use the `draft/extended-isupport` capability name to be interoperable with
+other software implementing a compatible work-in-progress version. The final
+version of the specification will use unprefixed capability names.
+
+## Introduction
+
+`RPL_ISUPPORT` is used to advertise metadata about the server as key-value
+entries. However, `RPL_ISUPPORT` is only sent by servers after connection
+registration. This undermines the usefulness of `RPL_ISUPPORT`: some of the
+metadata would be useful to clients prior to connection registration. This
+extension adds a mechanism to send `RPL_ISUPPORT` messages before connection
+registration, as well as an end of `RPL_ISUPPORT` list indication.
+
+## Implementation
+
+### `draft/extended-isupport` capability
+
+When the `draft/extended-isupport` capability is enabled by the client, the
+server MUST send one or more `RPL_ISUPPORT` messages (grouped inside a
+`draft/isupport` batch if the `batch` capability is enabled). The capability
+MAY be enabled by the client before connection registration completes (ie,
+before the client sends `CAP END`, and before the server sends `RPL_WELCOME`).
+
+Before connection registration completes, while `draft/extended-isupport` is
+enabled, the server MAY send updates to the key-value entries via subsequent
+`RPL_ISUPPORT` messages (the same way it would after connection registration
+completes without this extension).
+
+The server MAY skip the `RPL_ISUPPORT` replies usually sent when connection
+registration completes, if it already sent all information.
+
+Before connection registration, the server MAY send only a subset of the full
+`RPL_ISUPPORT` list. In that case, the server MUST send a `RPL_ISUPPORT` list
+when connection registration completes with entries previously omitted.
+
+If the client enables `draft/extended-isupport` after connection registration,
+the server MUST send one or more `RPL_ISUPPORT` messages. If the client
+enables, disables and then enables again `draft/extended-isupport` at any time,
+the server MUST send one or more `RPL_ISUPPORT` messages. If the client
+requests the `draft/extended-isupport` capability while it's already enabled,
+the server MUST NOT send any `RPL_ISUPPORT` messages in response.
```

Comment:
What's the purpose of this? Servers can already send RPL_ISUPPORT messages to notify changes without this cap. I don't see why enabling the cap should trigger isupport being sent again. Bear in mind clients can also trigger isupport being sent with the /version command.

Reply:
The goal is to have (1) all of ISUPPORT tokens get sent pre-connection-registration and (2) keep things as dumb/simple as possible. See also https://github.com/ircv3/ircv3-specifications/pull/543#discussion_r1613437758

Reply:
I get (1), but this para seems to be about post registration behaviour.

By (2) do you mean you want to avoid servers having to special case whether it's pre or post reg before deciding whether to sent isupport? If so, this para is worded as though it *is* a special case which might defeat the objective a bit.

Reply:
Yeah, the intent is to have the exact same behavior regardless of registration state...

Reply:
Tried to make this more explicit.